src/webgpu/api/operation/command_buffer/render/index_format.spec.ts

```diff
@@ -0,0 +1,299 @@
+export const description = `Index format tests.`;
+
+import { TestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import glslangModule from '../../../../util/glslang.js';
+
+export class IndexFormatTest extends GPUTest {
+  private glslang: any;
+  private pipeline!: GPURenderPipeline;
+  private colorAttachment!: GPUTexture;
+  private result!: GPUBuffer;
+  private swapChainFormat: GPUTextureFormat = 'bgra8unorm';
+
+  async initResources(format: GPUIndexFormat): Promise<void> {
+    // glslang module
+    this.glslang = await glslangModule();
+
+    // render pipeline
+    this.pipeline = this.MakeRenderPipeline(format);
+
+    const context = this.CreateRenderContext(true);
```

Comment:
Don't use a swap chain to get a render target; just use a regular createTexture. Take a look at some other tests on the glsl-dependent branch which test rendering results.

Reply:
Thanks, I will use the way in glsl-dependent branch.